app/lib/debounce.js

```diff
@@ -0,0 +1,53 @@
+// https://github.com/component/debounce/blob/master/index.js
+module.exports = (func, wait, immediate) => {
```

Comment:
This debouncer uses the task queue, not the microtask queue like the one in the blocks is implemented.